### PR TITLE
chore: Build with go 1.22

### PR DESCRIPTION
**Pull request by mrueg**
**What this PR does / why we need it**:
Moves go version from 1.21 to 1.22

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*
None
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ OS ?= $(shell uname -s | tr A-Z a-z)
 ALL_ARCH = amd64 arm arm64 ppc64le s390x
 PKG = github.com/prometheus/common
 PROMETHEUS_VERSION = 2.46.0
-GO_VERSION = 1.21.8
+GO_VERSION = 1.22.1
```
Comment (CatherineF-dev) [inline Makefile:19]: nit: will a higher 1.22.x reduce more vuls?

Comment (mrueg) [reply Makefile:19]: Yes, thanks. got 1.22.2 got released yesterday.

**Review by dgrisonnet — APPROVED** [review]
/lgtm
/approve

**Pull request merged**